### PR TITLE
Cast object parameters to string in Database\Database

### PR DESCRIPTION
**Pull request by MrPetovan**
Follow-up to #10722
Follow-up to #10817

### CODE REVIEW

File: src/Database/Database.php [hunk 1]
```diff
@@ -549,12 +549,14 @@ public function p($sql)
...
 				}
 
-				foreach ($args as $param => $value) {
+				foreach (array_keys($args) as $param) {
+					$data_type = PDO::PARAM_STR;
 					if (is_int($args[$param])) {
 						$data_type = PDO::PARAM_INT;
-					} else {
-						$data_type = PDO::PARAM_STR;
+					} elseif ($args[$param] !== null) {
```
Comment (annando) [inline src/Database/Database.php:556]: Do we have to handle `is_float` here as well?

File: src/Database/Database.php [hunk 2]
```diff
@@ -605,13 +607,16 @@ public function p($sql)
...
 				$values      = [];
-				foreach ($args as $param => $value) {
+				foreach (array_keys($args) as $param) {
 					if (is_int($args[$param])) {
 						$param_types .= 'i';
 					} elseif (is_float($args[$param])) {
 						$param_types .= 'd';
 					} elseif (is_string($args[$param])) {
 						$param_types .= 's';
+					} elseif (is_object($args[$param]) && method_exists($args[$param], '__toString')) {
```
Comment (annando) [inline src/Database/Database.php:617]: Is this conversion needed when there is the casting from line 557?

Comment (MrPetovan) [reply src/Database/Database.php:617]: Line 557 is the PDO case, this is the mysqli case.

Comment (annando) [reply src/Database/Database.php:617]: Ah okay.

**Pull request merged**